.github/workflows/ci.yml

```
@@ -0,0 +1,90 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  qa:
+    name: QA
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        toxenv:
+          - checkqa
+          - readme
+    steps:
+      - uses: actions/checkout@master
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+      - name: Set PY
+        run: echo "::set-env name=PY::$(python -VV | sha256sum | cut -d' ' -f1)"
+      - uses: actions/cache@v1
+        with:
+          path: ~/.cache/pre-commit
+          key: pre-commit|${{ env.PY }}|${{ hashFiles('.pre-commit-config.yaml') }}
+      - name: Install tox
+        run: pip install tox
+      - name: Test ${{ matrix.toxenv }}
+        run: tox -e ${{ matrix.toxenv }}
+
+  cpython:
+    name: Test
+    runs-on: ${{ matrix.os }}-latest
+    strategy:
+      matrix:
+        os:
+          - Ubuntu
+          - Windows
+        python-version:
+          - 2.7
+          - 3.5
+          - 3.6
+          - 3.7
+          - 3.8
+        pip-version:
+          - "20.0"
+          - "latest"
+    steps:
+      - uses: actions/checkout@master
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install tox
+        run: pip install tox
+      - name: Test pip ${{ matrix.pip-version }}
+        run: tox -e pip${{ matrix.pip-version }}-coverage
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v1.0.6
+        with:
+          file: ./coverage.xml
+          name: ${{ matrix.os }}-${{ matrix.python-version }}-${{ matrix.pip-version }}
```

In this context, there's also a `runner.os` which stands for `Linux`/`Windows`/`macOS` rather than your factors. Maybe use it instead?
```suggestion
          name: ${{ runner.os }}-${{ matrix.python-version }}-${{ matrix.pip-version }}
```

---

Seems reasonable.

---

But what's wrong with `Windows-3.8-20.0`? Do you suggest `windows-latest-3.8-20.0` instead?

---

No, it'd be just `Windows`. But it'll use the actual runner OS, not the name of _your_ matrix factor.

---

So it'll not depend on whether you rename things in the matrix

---

Got it, good point.

---

Addressed in 988dfbc.